components/Drawer/src/index.svelte

```diff
@@ -8,34 +9,64 @@
 	export let value = false;
 	export let target: null | HTMLElement = null;
 	export let cls = '';
-	export let attrs = {}
+	export let attrs = {};
+	export let footer = false;
+
 	const dispatch = createEventDispatcher();
+	$: showInner = value;
+
+	const close = () => {
+		showInner = !showInner;
+		dispatch('close', showInner);
+	};
+
 	const toggleClose = () => {
-		dispatch('close');
+		close();
+	};
+
+	const cancel = () => {
+		dispatch('cancel');
+		close();
 	};
 
-	$: isRight = (placement === 'right')
+	const confirm = () => {
+		dispatch('confirm');
+		close();
+	};
+
+	$: isRight = placement === 'right';
 </script>
+
 <KClientOnly>
 	<KMask {target} {value}>
-		<div class="k-drawer--base k-drawer--base__dark {isRight ? 'right-0' : 'left-0'} {cls}"
-			 {...attrs}
-			 out:fly={{ duration: 250, x: isRight ? 200 : -200 }}
-			 in:fly={{ duration: 250, x: isRight ? 200 : -200 }}>
-			<div class="k-drawer--op {isRight ? 'justify-start' : 'justify-end'}">
-				<KIcon
+		<div
+			class="k-drawer--base k-drawer--base__dark {isRight ? 'right-0' : 'left-0'} {cls}"
+			{...attrs}
+			out:fly={{ duration: 250, x: isRight ? 200 : -200 }}
+			in:fly={{ duration: 250, x: isRight ? 200 : -200 }}
+		>
+			<slot name="header">
+				<div class="k-drawer--op {isRight ? 'justify-start' : 'justify-end'}">
```

Comment:
How does the custom header expand and collapse the drawer? These behaviors should be controlled by the user in this case, and it will no longer trigger the close event, which should be explained in the documentation